ResidentInformationApi/Startup.cs

```diff
@@ -100,15 +102,46 @@ public static void ConfigureServices(IServiceCollection services)
                 if (File.Exists(xmlPath))
                     c.IncludeXmlComments(xmlPath);
             });
+
+            RegisterGateways(services);
             RegisterUseCases(services);
+            RegisterHTTPClients(services);
         }
 
         private static void RegisterUseCases(IServiceCollection services)
         {
-            if (services == null) throw new ArgumentNullException(nameof(services));
-            // your usecases here
+            services.AddScoped<IListContactsUseCase, ListContactsUseCase>();
+        }
+
+        private static void RegisterGateways(IServiceCollection services)
+        {
+            services.AddScoped<IAcademyInformationGateway, AcademyInformationGateway>();
+            services.AddScoped<IHousingInformationGateway, HousingInformationGateway>();
+            services.AddScoped<IMosaicInformationGateway, MosaicInformationGateway>();
         }
 
+        private static void RegisterHTTPClients(IServiceCollection services)
+        {
+            var academyUrl = Environment.GetEnvironmentVariable("ACADEMY_API_LIST_URL");
+            var housingUrl = Environment.GetEnvironmentVariable("HOUSING_API_LIST_URL");
+            var mosaicUrl = Environment.GetEnvironmentVariable("MOSAIC_API_LIST_URL");
+
+            services.AddHttpClient<IAcademyInformationGateway, AcademyInformationGateway>(a =>
+            {
+                a.BaseAddress = new Uri(academyUrl);
+            });
+
+            services.AddHttpClient<IHousingInformationGateway, HousingInformationGateway>(a =>
+            {
+                a.BaseAddress = new Uri(housingUrl);
+            });
+
+            services.AddHttpClient<IHousingInformationGateway, HousingInformationGateway>(a =>
+            {
+                a.BaseAddress = new Uri(mosaicUrl);
+            });
```

Comment:
I don't think you need to register the gateways twice, you can just register them with HttpClient.

Reply:
Does addhttpclient perform two functions: the same function as addscoped so the gateways are injected into the usecase and injects the httpclient into the gateway? 
I thought the addhttpclient just handled injecting the httpclient into the gateway?

Reply:
Yeah I believe it both scoped it and injects the HTTP client

Reply:
Yes, just checked they are registered as transient